packages/sql-mapper/lib/telemetry.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+function wrapQuery (app, db, request) {
+  const { startSpan, endSpan, SpanKind } = app.openTelemetry
+  async function wrappedFunction () {
+    const query = arguments[0]
+
+    let namePrefix, dbSystem
+    if (db.isPg) {
+      namePrefix = 'pg.query:'
+      dbSystem = 'postgresql'
+    } else if (db.isMySql) {
+      namePrefix = 'mysql.query:'
+      dbSystem = 'mysql'
+    } else if (db.isMariaDB) {
+      namePrefix = 'mariadb.query:'
+      dbSystem = 'mysql'
+    } else if (db.isSQLite) {
+      namePrefix = 'sqlite.query:'
+      dbSystem = 'sqlite'
+    } else {
+      namePrefix = 'db.query:'
+      dbSystem = 'unknown'
+    }
+
+    const format = {
+      escapeIdentifier: (str) => (str),
+      formatValue: (value, index) => ({ placeholder: `$${index + 1}`, value }),
+    }
+    const { text: queryText } = query.format(format)
+    const spanName = `${namePrefix}${queryText.replace(/\n|\r/g, '')}`
+
+    const ctx = request.span?.context
+
+    const telemetryAttributes = {
+      'db.statement': queryText,
+      'db.system': dbSystem,
+    }
+
+    let span
+    try {
+      span = startSpan(spanName, ctx, telemetryAttributes, SpanKind.CLIENT)
+      const result = await db.query.apply(db, arguments)
+      endSpan(span)
+      return result
+    } catch (err) /* istanbul ignore next */ {
+      endSpan(span, err)
+      throw err
+    }
+  }
+  return wrappedFunction
+}
+
+function wrapDB (app, db, request) {
+  const newDb = Object.create(db)
+  newDb.query = wrapQuery(app, db, request)
+  newDb.tx = async function wrapTx (func) {
+    return db.tx(async (db) => {
+      const _newDb = Object.create(db)
+      _newDb.query = wrapQuery(app, db, request)
+      return func(_newDb)
+    })
+  }
+  return newDb
+}
+
+const setupTelemetry = app => {
+  // Decorate the request with the wrapped DB.
+  // We need that for the queries written directly using `db`
+  app.addHook('onRequest', async (req, _reply) => {
+    const { db, sql } = app.platformatic
+    req['db'] = wrapDB(app, db, req)
+    req['sql'] = sql // For symmmetry, if we get the DB from the request, we should also get the SQL (even if unchanged)
+  })
```

Comment:
Please use a decorator with a getter instead of an hook. A hook adds overhead for every request, while a getter only on first use.